src/Yasumi/Filters/BankHolidaysFilter.php

```diff
@@ -43,12 +44,14 @@ public function accept(): bool
      */
     public function count(): int
     {
-        $days = \array_keys(\iterator_to_array($this));
+        $names = \array_map(static function (&$holiday) {
```

Comment:
This code block seems to occur in all the Filter classes. Do you think this could be refactored to an isolated code?

Reply:
Moved to new method `Holiday::count()`. I'm not sure what the use-case is for this kind of counting, but I guess it is equally useful as a utility method.

Reply:
We also use the counting in `AbstractProvider::count()`, so it is not specific to filters. But I could introduce an abstract filter class. Then at least it would only be duplicated twice. Do you prefer that?


--

> The use-case is straight forward in that each filter is able to return the total number filtered holidays. As substituted holidays don't need to be counted in the total number of holidays, [...]

This sounds like a very specific use-case (opinionated behaviour). For instance, in the following scenario, you would likely prefer a different definition:
```
print "Pay attention to these " . count($iterator) . " days:\n";
foreach ($iterator as $holiday) {
  print $holiday->format('Y-m-d') . ': ' . $holiday->getName() . "\n";
} 
```

I think it is rather surprising that `count($iterator)` does not reflect the number of `foreach` iterations.

But changing this would break BC, so it is not relevant for now. It was just to elaborate my comment about use-case :-)

Reply:
I was indeed thinking about an abstract filter class.

The `count($iterator)` correctly counts the number of iterations. The issue is actually when a holiday is being substituted, that holiday is counted twice (because a new Holiday instance is added to the internal array). 

The (domain) question is then: would you want to count the substituted holiday as well or should it be left out? At the time of building this, I felt a substituted holiday shouldn't be included in the count. :)

Reply:
I have added an abstract filter class.

Let's discuss the domain question some other day :-)